plugins/pi-genie/scripts/smoke.sh

```
@@ -0,0 +1,130 @@
+# Preconditions
+# ---------------------------------------------------------------------------
+[ -n "${PI_HOME:-}" ] || die "PI_HOME must be set to a throwaway pi home (never the real ~/.pi)"
+
+real_pi="$(cd "$HOME/.pi" 2>/dev/null && pwd -P || printf '%s' "$HOME/.pi")"
+[ "$(cd "$PI_HOME" 2>/dev/null && pwd -P || printf '%s' "$PI_HOME")" != "$real_pi" ] \
+  || die "refusing to run against the real PI_HOME ($PI_HOME)"
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+plugin_dir="$(cd "$script_dir/.." && pwd)"
+target="$PI_HOME/agent/extensions/genie"
+ext="$plugin_dir/extension.ts"
```

Shellcheck (0.11.0)

[warning] ext appears unused. Verify use (or export if used externally).

(SC2034)